### ISSUE TITLE
F.embedding has unexpected behavior with non-2d weights

### ISSUE THREAD

**Issue by shawnjhenry** [pytorch/pytorch#55185]
## 🐛 Bug

When calling F.embedding with a weight tensor that has > 2 dimensions, the behavior is unexpected. 

## To Reproduce

wgt = torch.randn(100, 32, 7, requires_grad=True)
idx = torch.randint(0, 100, (17, 5))
emb = F.embedding(idx, wgt)
cst = torch.sum(emb)
print(cst)
cst.backward()

## Expected behavior

If you run the above code, a result will be printed, but backward will throw a runtime invalid shape error.  If gradients are not supported for using F.embedding with weights with > 2 dimensions then calling forward should throw a dimension error.  Otherwise there's a bug in the implementation of backward.  

## Environment

 - PyTorch Version: 1.7
 - OS: Mac
 - How you installed PyTorch: conda
 - Build command you used (if compiling from source): n/a
 - Python version: 3.6.10
 - CUDA/cuDNN version: n/a
 - GPU models and configuration: n/a
 - Any other relevant information: n/a

cc @albanD @mruberry @jbschlosser

**Comment by jbschlosser** [comment]
Thanks for reporting this! I was able to reproduce it on master.

[F.embedding()](https://pytorch.org/docs/stable/nn.functional.html#embedding) is not intended to support non-2D weight, so we would accept a PR adding a dim check to ensure this.
